### ISSUE TITLE
notify user if scheduled backup went wrong

### ISSUE THREAD

**Issue by pdewouters** [humanmade/backupwordpress#767]
do scheduled backups fail silently? the problem is an error can go unnoticed for weeks until the user tries to use a backup to restore their site only to notice that something is missing.
we should try and report any errors or check the zip and make sure it's complete

**Comment by willmot** [comment]
Yeah they basically do, problem is if cron aren't working then the schedule code never runs.

We could write some code which checks the previous scheduled backups have run as expected that runs when the user visits the admin, but we couldn't email them at the time.

**Comment by willmot** [comment]
> we should try and report any errors or check the zip and make sure it's complete

We do catch errors and if the user has selected to have the backup emailed to them then we also email them backup errors. Perhaps we should just email everyone backup errors?

**Comment by pdewouters** [comment]
I'm down with that

**Comment by pdewouters** [comment]
I think we should ask permission before sending emails? I don't like adding new settings but it should be some kind of opt-in

**Comment by willmot** [comment]
I'm not so sure. If you're using a service and it fails you would expect to be notified. If you don't want to be notified of errors then you mustn't care that backups are failing, in-which case you can just opt-out by removing the plugin.

I think the fact that without notifications there are people who have backups failing but don't know about it is worse than the fact that people who don't want email would start getting them. If they don't want them they can remove the plugin.

Said differently I think these kinds of emails would be considered critical service emails and thus wouldn't need to be opt-in.

**Comment by pdewouters** [comment]
We should implement a notification service which allows different types of notifs such as email, slack message, push notif, sms,...

**Comment by willmot** [comment]
They would make good extensions actually, nice idea!
